### ISSUE TITLE
Suggestion: Trigger Preview

### ISSUE THREAD

**Issue by ExplodingBill** [absoIute/Mega-Hack-Pro-Future#270]
Trigger Preview would make it so that when a move trigger activates it will create a ghost copy of the object where it would be moved to:
  
![Screenshot (35)](https://user-images.githubusercontent.com/95105680/144927301-9b90b082-2fd8-4119-bdab-f1c7989ec1c6.png)

**Comment by Brittank88** [comment]
I feel like this suggestion should be broadened to a full system of previewing tools, however, I also feel that this might be better applied to BetterEdit, rather than directly into MHv7.

**Comment by ExplodingBill** [comment]
No, this would be like a gameplay hack so you can see this while you are playing so say, On Geometrical Dominator you would be able to see where all of the platforms are moving to:    ![Screenshot (37)](https://user-images.githubusercontent.com/95105680/145343217-db05d2e9-593e-43aa-825d-854a93c56069.png)

**Comment by absoIute** [comment]
it'd definitely be interesting to see but probably not something I'd look into for a while

so wonder if @HJfod likes the idea for betteredit

**Comment by HJfod** [comment]
I'm definitely up for adding this as an editor feature for BetterEdit. I think keeping general editor features in their own mod(s) that users can add/remove at will is better than cramming everything into one mod menu.

I'm not sure how this would be useful for gameplay. In the case provided (Geometrical Dominator) it could work, but as most levels nowadays utilize much more quantities of and much more complex movement, the visualizer would pretty much have to be powered by AI to provide actually useful information (i.e. show only moving blocks that matter, and where they are moving to). For the editor, where you can select what groups to isolate, I feel this would be more useful.

**Comment by absoIute** [comment]
Yeah that makes a lot more sense Mega Hack focuses a lot more on the playing experience so it is definitely more suited for BE. Since it'd be more an editor feature maybe it could also just show simple line paths like triggers w/ duration on do.

**Comment by HJfod** [comment]
The 'show movement line path' thing has been requested a ton ever since BE was started, I'm definitely going to add it but it requires figuring out how to actually calculate where to draw the line :P

**Comment by TheSillyDoggo** [comment]
> The 'show movement line path' thing has been requested a ton ever since BE was started, I'm definitely going to add it but it requires figuring out how to actually calculate where to draw the line :P

Here's a way you could do it,  Work out the centre position between the objects in the group id and make the line go up

**Comment by HJfod** [comment]
Yeah that isn't the problem though, the problem is when you have more than one move trigger targeting the same group at the same time and they have different easings, so you somehow have to figure out how to calculate where the line'll curve to :P I mean it's most likely just a matter of figuring out what the CC easing algorithms are and using those.

**Comment by ExplodingBill** [comment]
ok, if I find out what the easing algorithms are I'll post them in a Repository

**Comment by Brittank88** [comment]
> Yeah that isn't the problem though, the problem is when you have more than one move trigger targeting the same group at the same time and they have different easings, so you somehow have to figure out how to calculate where the line'll curve to :P I mean it's most likely just a matter of figuring out what the CC easing algorithms are and using those.

Well, you could technically bruteforce it by simulating the movement of an object and literally noting its path every frame or something and then drawing through those points. 

Would be better to leverage the actual algorithms though tbh. If you can find a way to compose the easing algorithms you could build functions that produce the right points to draw through without ever having to do a simulation, right?
